proton-c/bindings/python/proton/utils.py

```diff
@@ -132,10 +132,12 @@ def __init__(self, connection, receiver, fetcher, credit=1):
             raise LinkException("Failed to open receiver %s, source does not match" % self.link.name)
         if credit: receiver.flow(credit)
         self.fetcher = fetcher
+        self.container = connection.container
 
     def __del__(self):
         self.fetcher = None
-        self.link.handler = None
+        self.link.handler = None  # fails if reactor finalizes first
```

Comment:
+1 the comment should read something like "this would cause a core dump if the reactor finalized first, but that is impossible because *explain refcounting tricks here*" It is important to leave some explanation, this is exactly the sort of thing that someone in future will think "this doesn't seem necessary" and change it without realizing they are re-introducing a heisenbug.